api/v1alpha1/ovndbcluster_types.go

```diff
@@ -0,0 +1,164 @@
+type OVNDBClusterSpec struct {
+	ContainerImage string `json:"containerImage,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern="^NB|SB$"
+	// DBType - NB or SB
+	DBType string `json:"dbType"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Maximum=1
+	// +kubebuilder:validation:Minimum=0
+	// Replicas of OVN DBCluster to run
+	Replicas int32 `json:"replicas"`
+
+	// +kubebuilder:validation:Optional
+	// NodeSelector to target subset of worker nodes running this service
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=info
+	// LogLevel - Set log level info, dbg, emer etc
+	LogLevel string `json:"logLevel"`
+
+	// +kubebuilder:validation:Optional
+	// Resources - Compute Resources required by this service (Limits/Requests).
+	// https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// StorageClass
+	StorageClass string `json:"storageClass,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// StorageRequest
+	StorageRequest string `json:"storageRequest"`
+}
+
+// OVNDBClusterStatus defines the observed state of OVNDBCluster
+type OVNDBClusterStatus struct {
+	// ReadyCount of OVN DBCluster instances
+	ReadyCount int32 `json:"readyCount,omitempty"`
+
+	// Map of hashes to track e.g. job status
+	Hash map[string]string `json:"hash,omitempty"`
+
+	// Conditions
+	Conditions condition.Conditions `json:"conditions,omitempty" optional:"true"`
+
+	// RaftAddress -
+	RaftAddress string `json:"raftAddress,omitempty"`
+
+	// DBAddress -
+	DBAddress string `json:"dbAddress,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.conditions[0].status",description="Status"
+//+kubebuilder:printcolumn:name="Message",type="string",JSONPath=".status.conditions[0].message",description="Message"
+
+// OVNDBCluster is the Schema for the ovndbclusters API
+type OVNDBCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   OVNDBClusterSpec   `json:"spec,omitempty"`
+	Status OVNDBClusterStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// OVNDBClusterList contains a list of OVNDBCluster
+type OVNDBClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []OVNDBCluster `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&OVNDBCluster{}, &OVNDBClusterList{})
+}
+
+// IsReady - returns true if service is ready to server requests
+func (instance OVNDBCluster) IsReady() bool {
+	// Ready when:
+	// there is at least a single pod to serve the OVN DBCluster
+	return instance.Status.ReadyCount >= 1
+}
```

Comment:
Ditto.

Reply:
Not moved this method as it not pulling any extra dependencies and just relying on instance Status and will always we used with API object.